templates/flutter/lib/src/realtime_io.dart.twig

```diff
@@ -0,0 +1,90 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:math';
+import 'package:web_socket_channel/io.dart';
+import 'package:web_socket_channel/web_socket_channel.dart';
+import 'cookie_manager.dart';
+import 'realtime_subscription.dart';
+import 'realtime_base.dart';
+import 'realtime_mixin.dart';
+import 'client.dart';
+import 'client_io.dart';
+
+RealtimeBase createRealtime(Client client) => RealtimeIO(client);
+
+class RealtimeIO extends RealtimeBase with RealtimeMixin {
+
+  RealtimeIO(Client client) {
+    this.client = client;
+    this.getWebSocket = _getWebSocket;
+  }
+
+  Future<WebSocketChannel> _getWebSocket(Uri uri) async {
+    Map<String, String>? headers;
+    if (!(client as ClientIO).initialized) {
+      await (client as ClientIO).init();
+    }
+    final cookies = await (client as ClientIO).cookieJar.loadForRequest(uri);
+    headers = {HttpHeaders.cookieHeader: CookieManager.getCookies(cookies)};
+
+    final _websok = IOWebSocketChannel((client as ClientIO).selfSigned
+        ? await _connectForSelfSignedCert(uri, headers)
+        : await WebSocket.connect(uri.toString(), headers: headers));
+    return _websok;
+  }
+
+  /// Subscribe
+  ///
+  /// Use this method to subscribe to a channels and listen to
+  /// realtime events on those channels
+  RealtimeSubscription subscribe(List<String> channels) {
```

Comment:
We also need to allow passing a single string, that we will convert to an array 👍🏻

Reply:
@TorstenDittmann Dart doesn't allow multiple definitions of same function, will need a separate name is that a good idea? if yes, what would be the good name ?